src/errors.cpp

```diff
@@ -150,7 +140,28 @@ class security_category_t:
         if(code == cocaine::error::security_errors::token_not_found)
             return "specified token is not available";
 
-        return "cocaine.security error";
+        return cocaine::format("generic %s error", name());
+    }
+};
+
+struct locator_category_t: public std::error_category {
+    virtual
+    auto
+    name() const throw() -> const char* {
+        return "cocaine.service.locator";
+    }
+
+    virtual
+    auto
+    message(int code) const -> std::string {
+        if(code == cocaine::error::locator_errors::service_not_available)
```

Comment:
Why the fuck there are so much branches? Do we need them? Even more - without curly braces around the success statement.

The `switch` case looks better here.